### DISCUSSION TITLE
Question: How to activate mermaid from chunk* file?

### DISCUSSION THREAD

**Discussion by MODUSCarstenScholling** [dotnet/docfx#9089] [General]
Hello,

I've been using the classic template before. To get mermaid working, I had overwritten ```scripts.tmpl.partial``` like this:

```html
<script type="text/javascript" src="{{_rel}}styles/mermaid.min.js"></script>
<script>
	mermaid.initialize({startOnLoad: false});
	mermaid.init(undefined, ".lang-mermaid");
</script>
```

```markdown
'''mermaid
%%{
  init: {
    'theme': 'base',
    'themeVariables': {
      'primaryColor': '#00838F',
      'primaryTextColor': '#fff',
      'lineColor': '#A7ADB6'
    },
    "flowchart": {
      "htmlLabels": true,
      "curve": "linear" }
  }
}%%
flowchart TB
  ReportSelections("Report Selections")
  ReportConfiguration("Report Configuration")
  ReportSelections ==> ReportConfiguration
'''
```

With the modern template, the mermaid module is included in a chunk* file. Is there a way to tell docfx to activate it and handle the resulting `<div class="mermaid">`?

**Comment by yufeih** [comment]
mermaid diagrams are activated with `pre code.lang-mermaid` selector: https://github.com/dotnet/docfx/blob/main/templates/modern/src/markdown.ts#L41C54-L41C54

**Reply by MODUSCarstenScholling** [reply]
Yes, that works, thanks. But is there a way to force docfx to render a working

```html
<pre>
  <code class="lang-mermaid">
 ...
  </code>
</pre>
```

instead of the current

```html
<div class="mermaid">
 ...
</div>
```

from the markdown in the question?
Unfortunately the documentation on *specials* is somewhat limited...

**Comment by MODUSCarstenScholling** [comment]
# The solution

If you use markdig and set the docfx.config to use feature diagram (build/markdownEngineProperties/markdigExtensions/diagrams, then [a fenced code block with the mermaid language info will output a &lt;div class='mermaid'&gt; instead of a pre/code block](https://github.com/xoofx/markdig/blob/master/src/Markdig.Tests/Specs/DiagramsSpecs.md).

If the diagram extension is removed, then the internal handling automatically activates the mermaid drawing by building a pre/code.lang-mermaid block.
